src/imagery/i.eodag/i.eodag.py

```diff
@@ -215,54 +264,98 @@ def no_fallback_search(search_parameters, provider):
 
 
-def main():
-    # products: https://github.com/CS-SI/eodag/blob/develop/eodag/resources/product_types.yml
+def create_products_dataframe(eo_products):
+    result_dict = {"id": [], "time": [], "cloud_coverage": [], "product_type": []}
+    for product in eo_products:
+        if "id" in product.properties and product.properties["id"] is not None:
+            result_dict["id"].append(product.properties["id"])
+        else:
+            result_dict["id"].append(None)
+        if (
+            "startTimeFromAscendingNode" in product.properties
+            and product.properties["startTimeFromAscendingNode"] is not None
+        ):
+            try:
+                result_dict["time"].append(
+                    normalize_time(product.properties["startTimeFromAscendingNode"])
+                )
+            except:
+                result_dict["time"].append(
+                    product.properties["startTimeFromAscendingNode"]
+                )
+        else:
+            result_dict["time"].append(None)
+        if (
+            "cloudCover" in product.properties
+            and product.properties["cloudCover"] is not None
+        ):
+            result_dict["cloud_coverage"].append(product.properties["cloudCover"])
+        else:
+            result_dict["cloud_coverage"].append(None)
+        if (
+            "productType" in product.properties
+            and product.properties["productType"] is not None
+        ):
+            result_dict["product_type"].append(product.properties["productType"])
+        else:
+            result_dict["product_type"].append(None)
 
-    # setting the envirionmnets variables has to come before the dag initialization
-    setup_environment_variables()
+    df = pd.DataFrame().from_dict(result_dict)
+    return df
+
+
+def main():
+    # Products: https://github.com/CS-SI/eodag/blob/develop/eodag/resources/product_types.yml
 
     global dag
+    setup_environment_variables()
     dag = EODataAccessGateway()
-
     if options["provider"]:
         dag.set_preferred_provider(options["provider"])
     else:
-        gs.fatal(_("Please specify a provider..."))
-
-    # Download by ids... if ids are provided only these ids will be downloaded
-    if options["id"]:
-        ids = options["id"].split(",")
-        download_by_ids(ids)
+        # TODO: Add a way to search witout specifying a provider
+        gs.fatal(_("Please specify a provider."))
+
+    # Download by ids
+    # Searching for additional products won't take place
+    if options["id"] or options["file"]:
+        # Duplicates will be
+        ids_set = set()
+        if options["id"]:
+            ids_set.update(parse_id_option(options["id"]))
+        if options["file"]:
+            ids_set.update(parse_file_option(options["file"]))
+        gs.message(_("Found {} distinct product(s) IDs.".format(len(ids_set))))
+        for product_id in ids_set:
+            gs.message(product_id)
```

Comment:
If you feel the ids to download should be printed, I would rather do as follows:
```suggestion
        gs.message("\n".join(product_id))    
```